zmon-controller-app/src/main/java/org/zalando/zmon/api/SearchAPI.java

```diff
@@ -48,9 +59,32 @@
         List<QuickSearchResultItem> dashboards = searchService.quickSearchDashboards(search, teamList, limit);
         result.put("dashboards", dashboards);
 
-        List<QuickSearchResultItem> grafanaDashboards = searchService.quickSearchGrafanaDashboards(search, teamList, limit);
-        result.put("grafana_dashboards", grafanaDashboards);
+        result.put("grafana_dashboards", searchVisualizationDashboards(search, teams, limit, authHeader != null ? authHeader.split(" ")[1] : ""));
 
         return new ResponseEntity<>(result, HttpStatus.OK);
     }
+
+    private List<QuickSearchResultItem> searchVisualizationDashboards(String search, String teams, int limit, String token) {
+        List<QuickSearchResultItem> dashboards = new ArrayList<>();
+        Map<String, String> searchParams = new HashMap<>();
+        searchParams.put("query", search);
+        if (teams == null) {
+            searchParams.put("tag", "");
+        }
+        searchParams.put("tag", teams);
+        searchParams.put("limit", String.valueOf(limit));
```

Comment:
`limit` is a user provided value and is not validated before, so this line can throw an exception

Reply:
`limit` is primitive `int`
so I don't think any validation is required here